distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Delimited_Writer.enso

```diff
@@ -29,15 +32,45 @@ polyglot java import java.io.StringWriter
      operation. By default, a warning is issued, but the operation proceeds.
      If set to `Report_Error`, the operation fails with a dataflow error.
      If set to `Ignore`, the operation proceeds without errors or warnings.
-write_file : Table -> File_Format.Delimited -> File -> Existing_File_Behavior -> Problem_Behavior -> Any
-write_file table format file on_existing_file on_problems =
+write_file : Table -> File_Format.Delimited -> File -> Existing_File_Behavior -> Match_Columns -> Problem_Behavior -> Any
+write_file table format file on_existing_file match_columns on_problems =
     case on_existing_file of
         Existing_File_Behavior.Append ->
-            Errors.unimplemented "Appending to an existing File_Format.Delimited file is not implemented yet."
+            append_to_file table format file match_columns on_problems
         _ ->
             on_existing_file.write file stream->
                 write_to_stream table format stream on_problems related_file=file
 
+## PRIVATE
+   Handles appending to an existing file, ensuring that the columns are matched
+   against the ones already in the file.
+
+   If the file does not exist or is empty, it acts like a regular overwrite.
+append_to_file : Table -> File_Format.Delimited -> File -> Match_Columns -> Problem_Behavior -> Any
+append_to_file table format file match_columns on_problems =
+    Column_Name_Mismatch.handle_java_exception <| Column_Count_Mismatch.handle_java_exception <|
+        preexisting_headers = Delimited_Reader.detect_headers file format
```

Comment:
Not a big thing but we could avoid computing headers if in column count mode.

Reply:
I think this is one of these micro-optimizations which I'd avoid. The cost of computing the headers compared to the cost of all the file operations we perform while writing is minimal (especially as we still have to load the first row to know the column count anyway). And it would make the code more complicated because then I'd have to do split `detect_headers` function into two variants - one that just counts columns and one that reads the headers, possibly duplicating the code or making it more complex.

Reply:
fair enough. Didn't feel hugely more complicated but also wont make any real difference in performance.

Reply:
I think it's definitely doable and maybe not _hugely_ more complicated, but it adds _some_ complexity and so I thought it may just not be worth it, as having the code simpler and easier to understand is also a pretty big value.